src/Lifecycle.ts

```diff
@@ -602,7 +607,12 @@ async function doSetLoggedIn(credentials: IMatrixClientCreds, clearStorageEnable
     //
     // we fire it *synchronously* to make sure it fires before on_logged_in.
     // (dis.dispatch uses `window.setTimeout`, which does not guarantee ordering.)
-    dis.dispatch({ action: "on_logging_in" }, true);
+    //
+    // can be disabled to resolve "Cannot dispatch in the middle of a dispatch."
+    // error when it is invoked via another dispatch that is not yet finished.
+    if (dispatchOnLoggingIn) {
+        dis.dispatch({ action: "on_logging_in" }, true);
```

Comment:
`"on_logging_in"` doesn't seem to be used in the react-sdk or element-web code

Reply:
When creating new actions, please use the Action enum.

Reply:
I have added `"on_logging_in"` to Action, but this action wasn't created in this PR. I just added an `if` condition around it to fix the dispatcher.